### PR TITLE
General Table Result Parser

### PR DESCRIPTION
**Pull request by francinelapid**

### CODE REVIEW

File: lib/pavilion/plugins/results/table.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            name='table',
+            description="Parses tables"
+        )
+
+    def get_config_items(self):
+
+        config_items = super().get_config_items()
+        config_items.extend([
+            yc.StrElem(
+                'delimiter', required=True,
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/results/table.py]: This should default to whitespace.

File: lib/pavilion/plugins/results/table.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                help_text="Constant that will be placed in result."
+            ),
+            yc.StrElem(
+                'col_num', required=True,
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/results/table.py]: Should default to the number of column names if those are given.

File: lib/pavilion/plugins/results/table.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        )
+
+    def __call__(self, test, file, delimiter=None, col_num=None,
+                 row_col='', col_names = []):
+
+        match_list = []
+
+        # generate regular expression
+        value_regex = '(\S+| )'
+        new_delimiter = ' *' + delimiter + ' *'
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/results/table.py]: It's probably better to use '\s*' rather than ' *' here.

Comment (Paul-Ferrell) [reply lib/pavilion/plugins/results/table.py]: And in general.

File: lib/pavilion/plugins/results/table.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                'delimiter', required=True,
+                help_text="Constant that will be placed in result."
+            ),
+            yc.StrElem(
+                'col_num', required=True,
+                help_text="Number of columns in table, including row names, "
+                          "if there is such a column."
+            ),
+            yc.StrElem(
+                'row_col', default='False',
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/results/table.py]: This is a confusing name. Maybe 'has_header'?

File: lib/pavilion/plugins/results/table.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        new_delimiter = ' *' + delimiter + ' *'
+        value_regex_list = []
+        for i in range(int(col_num)):
+            value_regex_list.append(value_regex)
+        str_regex = new_delimiter.join(value_regex_list)
+        str_regex = '^ *' + str_regex + ' *$'
+
+        regex = re.compile(str_regex)
+        for line in file.readlines():
+            match_list.extend(regex.findall(line))
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/results/table.py]: `find_all` isn't really meant for this circumstance. You should just use regex.match.

File: lib/pavilion/plugins/results/table.py (hunk 2 above)
Comment (francinelapid) [reply lib/pavilion/plugins/results/table.py]: I check this logic in `_check_args`

File: lib/pavilion/plugins/results/table.py (hunk 5 above)
Comment (francinelapid) [reply lib/pavilion/plugins/results/table.py]: If there is a column for row names, the regex won't necessarily match the entire line.

Comment (Paul-Ferrell) [reply lib/pavilion/plugins/results/table.py]: You're right, I didn't read that code carefully enough.

**Pull request merged**